### PR TITLE
CJK overflow - update to match gl-shader

### PR DESCRIPTION
**Pull request by amyleew**
Per https://github.com/mapbox/mapbox-gl-native/issues/5873#issuecomment-238370996, working on updates to match https://github.com/mapbox/mapbox-gl-shaders/pull/22 and https://github.com/mapbox/mapbox-gl-js/pull/2930.

/cc @lbud @xrwang

### CODE REVIEW

**Comment by amyleew** [comment]
Looking good in Xcode! @brunoabinader can you take a final 👀 

screen shot 2016-08-08 at 6 55 10 pm

**Comment by friedbunny** [comment]
When it comes time to merge this, please squash and prefix the commit with `[core]`.

File: src/mbgl/shader/icon_shader.cpp [hunk 1]
```diff
@@ -21,8 +21,8 @@ void IconShader::bind(GLbyte* offset) {
     MBGL_CHECK_ERROR(glEnableVertexAttribArray(a_offset));
     MBGL_CHECK_ERROR(glVertexAttribPointer(a_offset, 2, GL_SHORT, false, stride, offset + 4));
 
-    MBGL_CHECK_ERROR(glEnableVertexAttribArray(a_data1));
-    MBGL_CHECK_ERROR(glVertexAttribPointer(a_data1, 4, GL_UNSIGNED_BYTE, false, stride, offset + 8));
+    MBGL_CHECK_ERROR(glEnableVertexAttribArray(a_texture_pos));
+    MBGL_CHECK_ERROR(glVertexAttribPointer(a_texture_pos, 2, GL_UNSIGNED_SHORT, false, stride, offset + 8));
```
Comment (lucaswoj) [inline src/mbgl/shader/icon_shader.cpp]: Do we need to define `a_texture_pos` in `shader.hpp`? https://github.com/mapbox/mapbox-gl-native/blob/69c529e26463303dea81a0b5d4b8540a9384d482/src/mbgl/shader/shader.hpp

**Comment by lucaswoj** [comment]
The only shaders with a `a_data1` attribute were `icon` and `sdf`. Because this PR removes that attribute in both cases, we need to make changes [everywhere in the codebase that `a_data1` or `a_data2` is referenced](https://github.com/mapbox/mapbox-gl-native/search?utf8=%E2%9C%93&q=a_data1&type=Code). 

 - [x] `src/mbgl/shader/icon_shader.cpp `
 - [x] `src/mbgl/shader/sdf_shader.cpp`
 - [ ] `src/mbgl/geometry/collision_box_buffer.cpp`
 - [ ] `src/mbgl/geometry/icon_buffer.cpp`
 - [ ] `src/mbgl/geometry/text_buffer.cpp`
 - [ ] `src/mbgl/shader/shader.hpp`
 - [ ] `src/mbgl/shader/shader.cpp`

Until all these are changed, I would expect segfaults, rendering errors, and complilation errors.

**Comment by amyleew** [comment]
Finally we got green 🍏 ✅ @lucaswoj @xrwang @jakepruitt !!!

Can't wait to :shipit: first thing tomorrow

**Comment by lucaswoj** [comment]
Was this ready to merge? I don't see a PR review from any GL Native core contributors or https://github.com/mapbox/mapbox-gl-native/pull/5908#issuecomment-238985186 having been addressed.

**Comment by amyleew** [comment]
Perhaps I misunderstood the PR review. Also I pushed `squash and commit` with the commit. https://github.com/mapbox/mapbox-gl-native/issues/5873 was taken care off, thought https://github.com/mapbox/mapbox-gl-native/issues/1681 was a GL native bug to fix seperately since this was happening with the original sized atlas as well.

**Comment by 1ec5** [comment]
#1681 is the original ticket tracking CJK glyph overflow issues.

**Pull request merged**